### PR TITLE
refactor(server): remove has asset

### PR DESCRIPTION
**Pull request by jrasm91**
Migrate album repository from `hasAsset` (removed) to `getAssetIds`.

### CODE REVIEW

File: server/src/services/album.service.ts [hunk 1]
```diff
@@ -146,8 +146,8 @@ export class AlbumService {
     const album = await this.findOrFail(id, { withAssets: true });
 
     if (dto.albumThumbnailAssetId) {
-      const valid = await this.albumRepository.hasAsset({ albumId: id, assetId: dto.albumThumbnailAssetId });
-      if (!valid) {
+      const results = await this.albumRepository.getAssetIds(id, [dto.albumThumbnailAssetId]);
```
Comment (michelheusschen) [inline server/src/services/album.service.ts:149]: `hasAsset` excluded deleted entities, while `getAssetIds` includes them. Will this still behave as expected?

Comment (jrasm91) [reply server/src/services/album.service.ts:149]: I guess that would technically let you set the thumbnail to an id that is in the album but deleted?

Comment (michelheusschen) [reply server/src/services/album.service.ts:149]: While that might not be ideal, web and mobile hide deleted assets in albums so you won't end up in this situation. The invalid thumbnail check for albums also doesn't exclude deleted assets, so it should be fine.

Comment (jrasm91) [reply server/src/services/album.service.ts:149]: That is a whole separate class of problems IMO. We might want to re-asses how deleted images are handled in albums, etc.

Comment (michelheusschen) [reply server/src/services/album.service.ts:149]: Yeah I agree, that's beyond the scope of this PR

**Review by danieldietzler — APPROVED** [review]

**Pull request merged**